### PR TITLE
Refactoring wavlib into 2 libs:

### PR DESCRIPTION
**Pull request by android-usb**
iolib - Streaming Player classes. Will be enhanced with streaming
recorders in the future.
parselib - parsers and loaders for WAV format data. Will be enhanced
with Writers and support for AIFF format.

### CODE REVIEW

File: samples/iolib/src/main/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+add_library( # Sets the name of the library.
+        iolib
+
+        # Sets the library as a static library.
+        STATIC
+
+        # Provides a relative path to your source file(s).
+        # io/stream
+        # ${CMAKE_CURRENT_LIST_DIR}/io/stream/FileInputStream.cpp
+        # ${CMAKE_CURRENT_LIST_DIR}/io/stream/InputStream.cpp
```
Comment (philburk) [inline samples/iolib/src/main/cpp/CMakeLists.txt]: can we remove these lines?

File: samples/parselib/src/main/cpp/CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # For more information about using CMake with Android Studio, read the
 # documentation: https://d.android.com/studio/projects/add-native-code.html
 
-# Sets the minimum version of CMake required to build the native library.
+# Sets the minimum version of CMake required toboe build the native library.
```
Comment (philburk) [inline samples/parselib/src/main/cpp/CMakeLists.txt]: covfefe

File: samples/parselib/src/main/cpp/CMakeLists.txt [hunk 3]
```diff
@@ -21,7 +21,7 @@ message("NDK is " ${NDK})
 set (OBOE_DIR ../../../../../)
 message("OBOE_DIR = " + ${OBOE_DIR})
 
-add_subdirectory(${OBOE_DIR} ./oboe-bin)
+#add_subdirectory(${OBOE_DIR} ./oboe-bin)
```
Comment (philburk) [inline samples/parselib/src/main/cpp/CMakeLists.txt]: does the parselib have any dependencies on Oboe?

File: samples/parselib/src/main/cpp/CMakeLists.txt [hunk 4]
```diff
@@ -53,9 +53,9 @@ add_library( # Sets the name of the library.
         ${CMAKE_CURRENT_LIST_DIR}/io/wav/WavRIFFChunkHeader.cpp
         ${CMAKE_CURRENT_LIST_DIR}/io/wav/WavStreamReader.cpp
         # player
-        ${CMAKE_CURRENT_LIST_DIR}/player/SampleBuffer.cpp
-        ${CMAKE_CURRENT_LIST_DIR}/player/OneShotSampleBuffer.cpp
-        ${CMAKE_CURRENT_LIST_DIR}/player/SimpleMultiPlayer.cpp
+        # ${CMAKE_CURRENT_LIST_DIR}/player/SampleBuffer.cpp
```
Comment (philburk) [inline samples/parselib/src/main/cpp/CMakeLists.txt]: can remove

File: samples/iolib/src/main/cpp/CMakeLists.txt (hunk 1 above)
Comment (android-usb) [reply samples/iolib/src/main/cpp/CMakeLists.txt]: Done.

File: samples/parselib/src/main/cpp/CMakeLists.txt (hunk 2 above)
Comment (android-usb) [reply samples/parselib/src/main/cpp/CMakeLists.txt]: Done

File: samples/parselib/src/main/cpp/CMakeLists.txt (hunk 4 above)
Comment (android-usb) [reply samples/parselib/src/main/cpp/CMakeLists.txt]: Done.

**Comment by android-usb** [comment]
All comments (and then some) addressed.

**Pull request merged**